src/utils/thread_helper.cpp

```diff
@@ -581,25 +582,27 @@ void ThreadHelper::AttachAllNodesTo(const std::vector<uint8_t> &aDatasetTlvs, Re
 
         if (hasActiveDataset)
         {
+            mAttachDelayMs            = kDelayTimerMilliseconds;
             mAttachHandler            = aHandler;
             mAttachPendingDatasetTlvs = datasetTlvs;
             ExitNow();
         }
         else
         {
-            aHandler(OT_ERROR_NONE);
+            aHandler(OT_ERROR_NONE, 0);
```

Comment:
Can we make the D-bus API async? For example, the handler is caller only when the BR becomes active (when the API call is truly success).

Reply:
Currently, when returned immediately after executing the d-bus command but we doesn't guarantee that the Border Router really did attach successfully. In the case we return `0` delay to the thread deamon, deamon has no idea when Border Router will become active, hence the mobile (GMS Core and iGHA) doesn't know when the Border Router becomes active. For iGHA, it has to wait for some time (how long?) for the Border Router becoming active before it can add the network to iOS, because iOS does check mDNS services when adding new networks.

iGHA doesn't have a good way to decide how long it need to wait before trying to add to iOS, so it has to try and backoff which is not elegant.

Reply:
Makes sense. I made the DBus call wait until it attaches then return.